File: packages/brownfield-navigation/ios/BrownfieldNavigationDelegate.swift
import Foundation
import React

@objc public protocol BrownfieldNavigationDelegate: AnyObject {
    @objc func navigateToSettings(_ user: UserType)
    @objc func navigateToReferrals(_ userId: String)
    @objc func requestNativeConfirmation(_ title: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock)
    @objc func showNativeBanner(_ message: String, onDismiss onDismiss: @escaping RCTResponseSenderBlock)
}

File: packages/brownfield-navigation/android/src/main/java/com/callstack/nativebrownfieldnavigation/BrownfieldNavigationDelegate.kt
package com.callstack.nativebrownfieldnavigation

import com.facebook.react.bridge.Callback
import com.facebook.react.bridge.Promise

interface BrownfieldNavigationDelegate {
  fun navigateToSettings(user: UserType)
  fun navigateToReferrals(userId: String)
  fun requestNativeConfirmation(title: String, promise: Promise)
  fun showNativeBanner(message: String, onDismiss: Callback)
}
